mewarpx/mewarpx/plotting.py

```diff
@@ -555,3 +553,44 @@ def get_2D_field_slice(data, xaxis, yaxis, slicevec=None, slicepos=None):
     if xaxis < yaxis:
         return dslice.T
     return dslice
+
+def get_figsize_from_warpx(max_dim=16.0, min_dim=0.0):
+    """Set up figure dimensions for outputs based on WARP aspect ratio.
+
+    Arguments:
+        max_dim (float, inches): Maximum size in either dimension.
+        min_dim (float, inches): Minimum size in either dimension. If supplied,
+            this takes precedence over the aspect ratio when the two conflict.
+
+    Returns:
+        figsize (tuple (xwidth, ywidth)): Figure size with given aspect ratio
+    """
+    if mwxrun.geom_str == 'RZ':
+        aspect_ratio = mwxrun.xmax / (mwxrun.zmax - mwxrun.zmin)
+    elif mwxrun.geom_str == 'XZ':
+        aspect_ratio = (mwxrun.xmax - mwxrun.xmin) / (mwxrun.zmax - mwxrun.zmin)
```

Comment:
Using `dx` and `dz` wouldn't really work either since those are generally equal, so we would get an aspect ratio of 1. The case where `dx != dz` is very rare for us, so I don't think we need to worry about it here.